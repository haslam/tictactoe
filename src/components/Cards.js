import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/card.module.css';

export default function Cards ({cards, handlePick}) {
  return (
    <div className={styles.cards}>
      {
        cards.map((card, index) => (
          <div className={styles.card} key={index} onClick={() => handlePick(index)}>
            {card}
          </div>
        ))
      }
    </div>
    
  )
}
Cards.propTypes = {
  cards: PropTypes.array.isRequired,
  handlePick: PropTypes.func.isRequired,
}
Cards.defaultProps = {
  cards: Array(9).fill(null),
}

