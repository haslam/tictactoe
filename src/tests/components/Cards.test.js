import React from 'react'
import { shallow } from 'enzyme'
import Cards from '../../components/Cards'

const testCards = Array(9).fill('X')
let wrapper, handlePick

beforeEach(() => {
  handlePick = jest.fn()
  wrapper = shallow(<Cards cards={testCards}
    handlePick={handlePick} />)
})

describe('Card', () => {
  it('should have 9 boxes', () => {
    expect(wrapper.children()).toHaveLength(9)
  })

  it('should ensure event is fired when clicked', () => {
    wrapper.children().forEach(node => {
      node.simulate('click')
      expect(handlePick).toHaveBeenCalled()
    })
  })

  it('should ensure component is rendered correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
