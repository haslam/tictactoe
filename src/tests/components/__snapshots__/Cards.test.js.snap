// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should ensure component is rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cards
    cards={
      Array [
        "X",
        "X",
        "X",
        "X",
        "X",
        "X",
        "X",
        "X",
        "X",
      ]
    }
    handlePick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
        <div
          className="card"
          onClick={[Function]}
        >
          X
        </div>,
      ],
      "className": "cards",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
          <div
            className="card"
            onClick={[Function]}
          >
            X
          </div>,
        ],
        "className": "cards",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
